packages-next/auth/src/index.ts

```diff
@@ -185,74 +180,61 @@ export function createAuth<GeneratedListTypes extends BaseGeneratedListTypes>({
    * Validates the provided auth config; optional step when integrating auth
    */
   const validateConfig = (keystoneConfig: KeystoneConfig) => {
-    const specifiedListConfig = keystoneConfig.lists[listKey];
-    if (keystoneConfig.lists[listKey] === undefined) {
-      throw new Error(
-        `A createAuth() invocation specifies the list "${listKey}" but no list with that key has been defined.`
-      );
+    const listConfig = keystoneConfig.lists[listKey];
+    if (listConfig === undefined) {
+      const msg = `A createAuth() invocation specifies the list "${listKey}" but no list with that key has been defined.`;
+      throw new Error(msg);
     }
 
     // TODO: Check for String-like typing for identityField? How?
-    const identityFieldConfig = specifiedListConfig.fields[identityField];
+    // TODO: Validate that the identifyField is unique.
+    // TODO: If this field isn't required, what happens if I try to log in as `null`?
+    const identityFieldConfig = listConfig.fields[identityField];
     if (identityFieldConfig === undefined) {
-      throw new Error(
-        `A createAuth() invocation for the "${listKey}" list specifies ${JSON.stringify(
-          identityField
-        )} as its identityField but no field with that key exists on the list.`
-      );
+      const i = JSON.stringify(identityField);
+      const msg = `A createAuth() invocation for the "${listKey}" list specifies ${i} as its identityField but no field with that key exists on the list.`;
+      throw new Error(msg);
     }
 
     // TODO: We could make the secret field optional to disable the standard id/secret auth and password resets (ie. magic links only)
-    const secretFieldConfig = specifiedListConfig.fields[secretField];
+    const secretFieldConfig = listConfig.fields[secretField];
     if (secretFieldConfig === undefined) {
-      throw new Error(
-        `A createAuth() invocation for the "${listKey}" list specifies ${JSON.stringify(
-          secretField
-        )} as its secretField but no field with that key exists on the list.`
-      );
+      const s = JSON.stringify(secretField);
+      const msg = `A createAuth() invocation for the "${listKey}" list specifies ${s} as its secretField but no field with that key exists on the list.`;
+      throw new Error(msg);
     }
-    const secretPrototype =
-      secretFieldConfig.type &&
-      secretFieldConfig.type.implementation &&
-      secretFieldConfig.type.implementation.prototype;
-    const secretTypename = secretFieldConfig.type && secretFieldConfig.type.type;
+
+    // FIXME: Until we explicity support user defined field types, should we just check against .type.type === 'Password'?
```

Comment:
are FIXME's and TODO's something that we reliably action in our workflow?
If yes, then lgtm, if no then we should either pair this with a matching JiRA issue, or ask the question directly in slack / confluence